### PR TITLE
[W5.11][T09-A3]Tey Xin Hui

### PR DESCRIPTION
**Pull request by TeyXinHui**
enhancement: add Birthday class

### CODE REVIEW

**Comment by TeyXinHui** [comment]
Hi, I would appreciate any advice on how to resolve that error

File: src/seedu/addressbook/data/person/Birthday.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+/**
+ * Represents a Person's birthday in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidBirthday(String)}
+ */
+public class Birthday {
+
+    public static final String EXAMPLE = "08-07-1890";
+    public static final String MESSAGE_BIRTHDAY_CONSTRAINTS = "Person birthdays format should be in DD-MM-YYYY, where DD means day, MM means month, YYYY means year";
+    public static final String BIRTHDAY_VALIDATION_REGEX = "^(?:(?:31(\\/|-|\\.)(?:0?[13578]|1[02]))\\1|(?:(?:29|30)(\\/|-|\\.)(?:0?[1,3-9]|1[0-2])\\2))(?:(?:1[6-9]|[2-9]\\d)?\\d{2})$|^(?:29(\\/|-|\\.)0?2\\3(?:(?:(?:1[6-9]|[2-9]\\d)?(?:0[48]|[2468][048]|[13579][26])|(?:(?:16|[2468][048]|[3579][26])00))))$|^(?:0?[1-9]|1\\d|2[0-8])(\\/|-|\\.)(?:(?:0?[1-9])|(?:1[0-2]))\\4(?:(?:1[6-9]|[2-9]\\d)?\\d{2})$";
+    //BIRTHDAY_VALIDATION_REGEX adapted from StackOverflow, implements leap year validation
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Birthday.java:17]: Good that you validate the date. however, I don't think you need to mention StackOverflow here.

File: src/seedu/addressbook/data/person/Birthday.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Represents a Person's birthday in the address book.
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Birthday.java:9]: You can omit the `in the address book.`

File: src/seedu/addressbook/data/person/Birthday.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                || (other instanceof Birthday // instanceof handles nulls
+                && this.value.equals(((Birthday) other).value)); // state check
+    }
+
+    @Override
+    public int hashCode() {
+        return value.hashCode();
+    }
+
+    public boolean isPrivate() {return this.isPrivate;}
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Birthday.java:59]: I think this code violates our coding standard..

File: src/seedu/addressbook/data/person/ReadOnlyPerson.java [hunk 4]
```diff
@@ -87,6 +94,9 @@ default String getAsTextHidePrivate() {
         }
         if (!getAddress().isPrivate()) {
             builder.append(" Address: ").append(getAddress());
+        }
+		if (!getBirthday().isPrivate()) {
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/ReadOnlyPerson.java:98]: Please fix the indentation by changing the tab to 4 spaces.

File: test/input.txt [hunk 5]
```diff
@@ -28,31 +28,32 @@
...
   add wrong args wrong args
-  add Valid Name 12345 e/valid@email.butNoPhonePrefix a/valid, address
-  add Valid Name p/12345 valid@email.butNoPrefix a/valid, address
-  add Valid Name p/12345 e/valid@email.butNoAddressPrefix valid, address
-  add Valid Name p/12345 e/valid@email.butNoTagPrefix a/valid, address t/goodTag noPrefixTag
+  add Valid Name 12345 e/valid@email.butNoPhonePrefix a/valid, address b/19-09-1990
+  add Valid Name p/12345 valid@email.butNoPrefix a/valid, address b/19-09-1990
+  add Valid Name p/12345 e/valid@email.butNoAddressPrefix valid, address b/19-09-1990
+  add Valid Name p/12345 e/valid@email.burNoBirthdayPrefix a/valid, address 19-09-1990
+  add Valid Name p/12345 e/valid@email.butNoTagPrefix a/valid, address b/19-09-1990 t/goodTag noPrefixTag
```
Comment (jeffryhartanto) [inline test/input.txt:35]: Good that you update the IO test.

**Comment by jeffryhartanto** [comment]
@TeyXinHui Some comments added. Please close the PR after reading comments.

BTW, this PR is supposed to be merge with the other W5.11 PR. Anyway, for this one you can leave it as is.

**Pull request closed without merging**